kratos/geometries/coupling_geometry.h

```diff
@@ -0,0 +1,357 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Tobias Teschemacher
+//  Contributors:    Pooyan Dadvand
+//                   Philipp Bucher
+//
+
+#if !defined(KRATOS_COUPLING_GEOMETRY_H_INCLUDED )
+#define  KRATOS_COUPLING_GEOMETRY_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "geometry.h"
+
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+/**
+ * @class CouplingGeometry
+ * @ingroup KratosCore
+ * @brief The CouplingGeometry can connect different geometries, those
+ can be of different entities but have to coincide with the dimension.
+ */
+template<class TPointType> class CouplingGeometry
+    : public Geometry<TPointType>
+{
+public:
+    ///@}
+    ///@name Type Definitions
+    ///@{
+
+    /**
+     * Geometry as base class.
+     */
+    typedef Geometry<TPointType> BaseType;
+    typedef Geometry<TPointType> GeometryType;
+
+    typedef typename GeometryType::Pointer GeometryPointer;
+
+    /**
+     * Pointer definition of CouplingGeometry
+     */
+    KRATOS_CLASS_POINTER_DEFINITION( CouplingGeometry );
+
+    typedef TPointType PointType;
+
+    typedef typename BaseType::IndexType IndexType;
+    typedef typename BaseType::SizeType SizeType;
+
+    typedef typename BaseType::PointsArrayType PointsArrayType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor for coupling one master to one slave geometry.
+    CouplingGeometry(
+        GeometryPointer pMasterGeometry,
+        GeometryPointer pSlaveGeometry)
+        : BaseType(PointsArrayType(), &(pMasterGeometry->GetGeometryData()))
+    {
+        KRATOS_DEBUG_ERROR_IF(pMasterGeometry->Dimension() != pSlaveGeometry->Dimension())
+            << "Geometries of different dimensional size!" << std::endl;
+
+        mpGeometries.resize(2);
+
+        mpGeometries[0] = pMasterGeometry;
+        mpGeometries[1] = pSlaveGeometry;
+    }
+
+    explicit CouplingGeometry(const PointsArrayType& ThisPoints)
```

Comment:
is this a valid use of the coupling-geomtery?

Reply:
The coupling geometry byitself does not contain the container of the Points - but to keep the interface this function is added. Thus, I guess it is correct.

Reply:
If you cannot support a constructor, please do not define it. It would be misleading.